### DISCUSSION TITLE
New projection starting from checkpoint and not from start

### DISCUSSION THREAD

**Discussion by shashikantpawar21** [Eventuous/eventuous#156] [Q&A] [answered]
services.AddSubscription<AllStreamSubscription, AllStreamSubscriptionOptions>(
"MainProjection",
builder => builder
**.UseCheckpointStore()**
.AddEventHandler<SecondaryProjection>()
**.AddEventHandler<<ThirdProjection>>()**  --> Added to existing thing
.WithPartitioningByStream(2));

When I added ThirdProjection / event handler , It start listening to new event maybe because of the checkpoint store ?
But I wanted it to listen to stream from the start of Main Projection only for this projection and not for existing projection.

Can you suggest something here ?

**Comment by shashikantpawar21** [comment]
This is urgent and concerning issue 
Can someone help me here?

**Comment by alexeyzimarev** [comment, marked as answer]
Subscriptions start from the checkpoint stored in the checkpoint store by subscription id. Adding a new event handler does not force the subscription to replay all the events for that particular handler; it will continue from the last know checkpoints. It's all described in the docs.

If you add a new handler that needs to start from scratch, you either need to replay everything by deleting the subscription checkpoint or create a new subscription for that handler alone.

**Reply by bartelink** [reply]
>or create a new subscription for that handler alone.

(when it's reached the tail, you can then collapse the handler into the other one, if that still makes sense)

**Reply by alexeyzimarev** [reply]
Yeah, that was implicit from my side :)

I can think of something to make it work, as it could be useful, but I am not sure when. It would require changing the checkpointing quite substantially as every handler would need to have its own checkpoint.

**Reply by bartelink** [reply]
Yes definitely wasn't expanding that for you, Alexey!
I'd say in general you don't want to be creating checkpoints/consumer groups per handler as that's not free - better to default to having groups that you can move handlers between and/or reset as a batch and then let people put things into groups of one if they need specific control for something
